### PR TITLE
Deploy Sequoia runner, 🔪 Monterey

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This will allow us to avoid the loss of bottles to Homebrew/core PRs.

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]
Thanks @carlocab! Unclear from the code but: we should only do this for Apple Silicon on Sequoia, not Intel. Intel bottles should only be manually dispatched and not added into the normal rotation.

**Comment by carlocab** [comment]
> we should only do this for Apple Silicon on Sequoia, not Intel. Intel bottles should only be manually dispatched and not added into the normal rotation.

Yes, already handled in

https://github.com/Homebrew/brew/blob/c5d09b411776547bdbc8e0435deaf690ca632413/Library/Homebrew/github_runner_matrix.rb#L184

**Comment by MikeMcQuaid** [comment]
@carlocab Great, thanks, ship away!

**Comment by fxcoudert** [comment]
Actually, I think we're in good shape now to deploy the ARM Sequoia runner on all homebrew-core PRs. We usually do it at around 60% of formulas bottled, and now we're at 57%. Moreover, we have relatively few breakages (and they're often easy to fix with clang flags) so it should be fine at this point.

**Comment by carlocab** [comment]
Cool, will adjust. Note: may take me until tomorrow to refactor out test failures nicely. (Or I can hardcode more assumptions into tests if we're in a hurry.)

File: Library/Homebrew/global.rb
Comment (Bo98) [inline Library/Homebrew/global.rb:61]: This is kinda weird global, especially if it's only for tests. I think the tests should just do `GitHubRunnerMatrix::NEWEST_HOMEBREW_CORE_MACOS_RUNNER`

Comment (carlocab) [reply Library/Homebrew/global.rb:61]: Sure, works for me. Just thought these values would be easier for other maintainers to tweak if they lived in global.

**Review by Bo98 — APPROVED** [review]
LGTM. I'll let you judge when best to merge this, bearing in mind 4.3.21 which dropped the "no bottle available" error for macOS 12 can take up to 24 hours to reach everyone (on default settings).

**Comment by carlocab** [comment]
> LGTM. I'll let you judge when best to merge this, bearing in mind 4.3.21 which dropped the "no bottle available" error for macOS 12 can take up to 24 hours to reach everyone (on default settings).

Cool, let's wait till tomorrow then

**Review by p-linnane — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Review by fxcoudert — APPROVED** [review]

**Comment by dduugg** [comment]
is this merge the cause of this job failure? https://github.com/Homebrew/brew/actions/runs/10854929440/job/30126441752?pr=18305

**Comment by carlocab** [comment]
I think that'll have been https://github.com/Homebrew/brew/pull/18296 and the latest tag.

**Comment by mozeryansky** [comment]
Am I the only one still using macOS 12? :(

**Comment by p-linnane** [comment]
> Am I the only one still using macOS 12? :(

No, but Monterey is officially unsupported by Apple. We only support the 3 most recent versions of macOS.

**Pull request merged**
